students/Steven/lesson05/mailroom03.py

```diff
@@ -0,0 +1,101 @@
+#! bin/user/env python3
+import sys
+from operator import itemgetter
+"""
+Program Part2:
+Update your mailroom program to:
+
+Use dicts where appropriate.
+See if you can use a dict to switch between the user’s selections.
+See if you can use a dict to switch between the users selections. see Using a Dictionary to switch for what this means.
+Convert your main donor data structure to be a dict.
+Try to use a dict and the .format() method to produce the letter as one big template,
+rather than building up a big string that produces the letter in parts.
+"""
+
+# Donor name, donation amount, donation history
+donor_list = {
+    "John Doe": [2000, 2],
+    "Jane Doe": [3000, 3],
+    "George Washington": [1, 1],
+    "Andrew Jackson": [20, 2],
+    "Benjamin Franklin": [100, 3]
+}
+
+# Main options menu for user to pick from
+def main():
+    while True:
+        options()  # Display each option to the user
+        try:
+            choice = int(input("What would you like to do? "))
+            # Using a dictionary for as a switch case
+            switch_func_dict = {
+                1: thank_you,
+                2: report,
+                3: all_letter,
+                4: quit
+            }
+            switch_func_dict.get(choice)()
+        except ValueError as error_message:
```

Comment:
👍 - a nice thing to do is to add a user friendly message here `print("Invalid menu option")`


what happens when they enter 5 ? (it will crash your program) this should be caught as well